masq/src/non_interactive_mode.rs

```diff
@@ -31,34 +32,31 @@ impl Default for Main {
 impl Main {
     pub fn new() -> Self {
         Self {
-            non_interactive_clap_factory: Box::new(NonInteractiveClapFactoryReal),
+            non_interactive_clap_factory: Box::new(NIClapFactoryReal),
             command_factory: Box::new(CommandFactoryReal),
             processor_factory: Box::new(CommandProcessorFactoryReal),
         }
     }
 
     fn extract_subcommand(args: &[String]) -> Option<Vec<String>> {
-        if args.len().eq(&1) {
-            return None;
-        }
-        let vec_candidate = args
-            .to_vec()
-            .into_iter()
-            .skip(1 + Self::eliminate_pre_subcommand_items(args[1].as_str()))
-            .collect::<Vec<String>>();
-        if !vec_candidate.is_empty() {
-            Some(vec_candidate)
+        let original = args.iter();
+        let one_item_shifted_forth = args.iter().skip(1);
+        let index = original.zip(one_item_shifted_forth).enumerate().find(
```

Comment:
Don't call this "index," because "index" means a number. Either call it "index_pair" or something similar, or put a .0 at the end of the find()